t/plugin/authz-casbin.t

```diff
@@ -0,0 +1,358 @@
+            local plugin = require("apisix.plugins.authz-casbin")
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/plugin_metadata/authz-casbin',
+                ngx.HTTP_PUT,
+                [[{
+                    "model": "[request_definition]
+                    r = sub, obj, act
+
+                    [policy_definition]
+                    p = sub, obj, act
+
+                    [role_definition]
+                    g = _, _
+
+                    [policy_effect]
+                    e = some(where (p.eft == allow))
+
+                    [matchers]
+                    m = (g(r.sub, p.sub) || keyMatch(r.sub, p.sub)) && keyMatch(r.obj, p.obj) && keyMatch(r.act, p.act)",
+
+                    "policy": "p, *, /, GET
+                    p, admin, *, *
+                    g, alice, admin"
+                }]]
+                )
+
+            local conf = {
+                username = "user"
+            }
+            local ok, err = plugin.check_schema(conf)
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- response_body
+done
+--- no_error_log
+[error]
+
+
+
+=== TEST 5: enable authz-casbin by Admin API
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "authz-casbin": {
+                            "username" : "user"
+                        }
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1982": 1
+                        },
+                        "type": "roundrobin"
+                    },
+                    "uri": "/*"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: no username header passed
+--- request
+GET /hello
+--- error_code: 403
+--- response_body_like eval
+qr/"Access Denied"/
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: username passed but user not authorized
+--- request
+GET /hello
+--- more_headers
+user: bob
+--- error_code: 403
+--- response_body
+{"message":"Access Denied"}
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: authorized user
+--- request
+GET /hello
+--- more_headers
+user: admin
+--- error_code: 200
+--- response_body
+hello world
+--- no_error_log
+[error]
+
+
+
+=== TEST 9: authorized user (rbac)
+--- request
+GET /hello
+--- more_headers
+user: alice
+--- error_code: 200
+--- response_body
+hello world
+--- no_error_log
+[error]
+
+
+
+=== TEST 10: unauthorized user before policy update
```

Comment:
Need a similar test for the policy_path configuration.

Reply:
Added the tests for enforcer from files.